test/test_torch.py

```diff
@@ -2341,6 +2341,13 @@ def test_zeros(self):
         torch.zeros(100, 100, out=res2)
         self.assertEqual(res1, res2)
 
+        res1 = torch.zeros(2, 2, dtype=torch.bool)
+        expected = torch.tensor([[False, False], [False, False]], dtype=torch.bool)
+        self.assertEqual(res1, expected)
+
+        halfTensor = torch.zeros(1, 1, dtype=torch.half)
+        self.assertEqual(halfTensor, torch.tensor([[0.]], dtype=torch.float16))
```

Comment:
The tests here are weirdly asymmetric with the test above. I... guess it's OK? It looks kind of weird though

Reply:
(The asymmetry is, the test above is testing that `out=` and non-out implementations match; and you're testing that tensor interprets bool literals in the same way that zeros fills out. I don't mind that much, because the name of the test suggests "general tests for zeros" which, OK, fine, why not.)